### PR TITLE
Fix/nav links dropdown

### PR DESCRIPTION
**Pull request by 0xRowdy**
## GitHub Issue

Finish SubNav Dropdown[#501](https://github.com/HausDAO/daohaus-monorepo/issues/501)

## Changes

A summary of the changes in the code in this PR. If making any UI changes consider adding screenshots or gifs.

Screen Shot 2022-07-22 at 4 16 58 PM

## Packages Added

No new packages, some refactoring of other components.

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: libs/ui/src/components/atoms/Link/Link.tsx [hunk 1]
```diff
@@ -25,14 +38,25 @@ export const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
           ref={ref}
         >
           {children}
-          {externalIcon && <RiExternalLinkLine />}
+          {linkType === 'external' && Icon ? <Icon /> : <RiExternalLinkLine />}
         </ExternalLink>
       );
     }
 
+    if (linkType === 'navigation') {
```
Comment (jordanlesich) [inline libs/ui/src/components/atoms/Link/Link.tsx:46]: Inherit in subnav. Probably won't be used outside of subnav

File: libs/ui/src/components/molecules/SubNav/SubNav.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+export type SubNavProps = {
+  className?: string;
+  navLinks?: NavLinkType[];
+  dropdownTriggerText?: string;
+  dropdownMenuSpacing?: string;
+  dropdownMenuAlign?: Align;
+  dropdownLinks?: NavLinkType[];
+};
+
+/*REVIEW:
```
Comment (jordanlesich) [inline libs/ui/src/components/molecules/SubNav/SubNav.tsx:25]: Yes

**Review by jordanlesich — APPROVED** [review]
Down for name change to <NavBar />.

Did this review while chatting w/ Rowdy.

File: libs/ui/src/components/atoms/Link/Link.tsx [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
+import { InternalLink, ExternalLink, NavLink } from './Link.styles';
 
 export interface LinkProps extends React.ComponentPropsWithRef<'a'> {
   href?: string;
-  externalIcon?: boolean;
+  Icon?: IconType;
+  isExternal?: boolean;
+  selected?: boolean;
+  disabled?: boolean;
+  linkType?: 'internal' | 'external' | 'navigation';
```
[the remark this answers is not in the document]
Comment (0xRowdy) [reply libs/ui/src/components/atoms/Link/Link.tsx:14]: Moved nav styled variant into SubNav

**Comment by 0xRowdy** [comment]
Decided to keep Mobile variant and Dropdown refactor in branch for Create In-app Navigation Component[#254](https://github.com/HausDAO/daohaus-monorepo/issues/254). Should have had the PR re-reviewed here if incorporated. This is better to keep PR smaller and update the references used in apps before refactor.

**Pull request merged**